docs: improve readme (#12)
metallang/metal: README.md

# Metal

> Prototype compiled language built in Rust with LLVM. We'll add a catchy slogan later.

Reviewer: If you feel like it, feel free to remove the latter sentence, no need to make compromises here :D

## Contributing

> Currently, Metal is more or less under source availability. Metal will eventually

Author: ```suggestion
Currently, Metal is unlicensed and solely owned by Vincent and Lena the maintainers. Metal will eventually open up and
```

> Currently, Metal is unlicensed and solely owned by Vincent and Lena the maintainers. Metal will eventually open up and

Reviewer: > owned by <us>
https://choosealicense.com/no-permission/

So say that i believe we'll at the very least need a LICENSE file that says "Copyright <us>. All rights reserved"

> become fully open source, but for now, it's far too undeveloped to require an open source license.

Reviewer: The "too underdeveloped to require an OSS licence" part sounds strange to me. No matter the degree of "developedness" of a piece of software, there's no point at which it requires to be OSS, right? Also, i think "source availability" suggests more freedom than absence of a license means (if not for GitHub's ToS, no one could even legally *view* the code). I propose leaving out this section until we decide on a license

> become fully open source, but for now, it's far too undeveloped to require an open source license.

Author: ```suggestion
become fully open source, but for now, since we're the only ones
y
```

## Development

Use nightly Rust.

> If you want to add a dependency, add it to the workspace Cargo.toml and then in the crate that you want to use this dependency in specify that dependency as dep_name = { workspace = true }. This makes updating deps much easier.

Reviewer: Why did you remove backticks?

Author: Byproduct of accidentally removing it then adding it back

# Metal

Prototype compiled language built in Rust with LLVM.

## Contributing

Currently, Metal is currently not open source software. Metal will eventually open up and

> become fully open source, but for now, since we're the only ones using this language it's fully closed.

Reviewer: ```suggestion
become fully open source, but for the time being, it's not open for contributions.
```

## Development

Use nightly Rust.

If you want to add a dependency, add it to the workspace Cargo.toml and then

> in the crate that you want to use this dependency in specify that dependency as dep_name = { workspace = true }.

Reviewer: ```suggestion
in the crate that you want to use this dependency in specify that dependency as `dep_name = { workspace = true }`.
```
